remove maxProcessingTime

nsauro opened the pull request:
fixes #96 

`maxProcessingTime` was only partially removed in my last PR where I changed the `QueueProcessor` / `Worker` / `Routee` relationship.  

Currently, the value is still being used and scheduled by the QueueProcessor, however I had only removed the case from the `Receive` function which responded to it.  This removes the last vestiges of it: removing it from the `ProcessingWorkerPoolSettings` class and removing the message class `QueueMaxProcessTimeReached`

kailuowang commented:
It's also present in reference.conf
and seems that PoolSize metric test failed again?

nsauro commented:
same with AutoScalingSpec, but that one seems to be a race condition:

https://github.com/iheartradio/kanaloa/blob/master/core/src/test/scala/kanaloa/reactive/dispatcher/queue/AutoScalingSpec.scala#L175-L176

nsauro commented:
I'll remove `maxProcessingTime` from the config, and see if i can stabilize these tests

kailuowang commented:
The problem with that `AutoScalingSpec` I think was that it's not certain which termination message comes first, I think. To fix it might need to use `expectMsgType[Terminated]` and then assert on the ref in that message is either `processor` or `a`,  rather than `expectTerminated`.

nsauro commented:
Ok, I removed that property from config.  I also added `eventually` checks on all tests which referenced `receivedMetrics` in `MetricCollectorScope`.  I added a different way of asserting that both the queueProcessor and autoScaler terminated in that failing AutoScalingSpec

nsauro commented on core/src/test/scala/kanaloa/reactive/dispatcher/queue/AutoScalingSpec.scala:
```diff
@@ -169,11 +169,23 @@ class AutoScalingSpec extends SpecWithActorSystem with OptionValues with Eventua
       watch(processor)
-      val a = system.actorOf(AutoScaling.default(processor, AutoScalingSettings(), MetricsCollector(None)))
-      watch(a)
+      val autoScaler = system.actorOf(AutoScaling.default(processor, AutoScalingSettings(), MetricsCollector(None)))
```
a bit more code than I would have liked to write, but I think it does the trick

kailuowang commented on core/src/test/scala/kanaloa/reactive/dispatcher/queue/AutoScalingSpec.scala:
```diff
@@ -169,11 +169,23 @@ class AutoScalingSpec extends SpecWithActorSystem with OptionValues with Eventua
+        case Terminated(`processor`)  ⇒ processorTerminated = true
+      }
+      autoScalerTerminated shouldBe true
+      processorTerminated shouldBe true
```
How about

``` Scala
List(expectMsgType[Terminated].actor, expectMsgType[Terminated].actor) should contain allOf (processor, autoScaler)
```

nsauro replied:
way better. will make the change

kailuowang commented on core/src/test/scala/kanaloa/reactive/dispatcher/queue/QueueSpec.scala:
```diff
@@ -207,8 +187,9 @@ class CircuitBreakerSpec extends SpecWithActorSystem {
 
       delegatee.expectNoMsg(30.milliseconds) //give some time for the circuit breaker to kick in
```
should remove this line as well, `eventually` should take care of it.

nsauro commented:
I reinstated that test I commented out.  It still doesn't work, but that is due to how Shutdown changed with last refactor.  I plan to change all this when I rethink shutdown a bit.

Looks like there is still some instability with some of the specs

kailuowang commented:
finally 👍
